### PR TITLE
Docs Improvement: Update Heroku Entry

### PR DESCRIPTION
**Pull request by gdjohnson**
When revamping our Commonwealth package ReadMe, as part of our [Onboarding Docs](https://github.com/hicommonwealth/commonwealth/issues/4999) push, I purged the Heroku & DataDog sections. Per consultation with Timothee, much of this information was (1) out of date (2) not especially relevant to new engineering hires.

However, rather than leave these areas undocumented, Tim & I have been meeting to work through how we use these tools, creating new knowledge base entries with up-to-date documentation. This is the first installment (Heroku); DataDog will likely come sometime next week.

Down the line (aspirationally), we would also have documentation our existing Heroku configuration, so that we could set up a new instance from scratch should the need arise. We're not there yet—I'd be curious to hear whether @jnaviask thinks this would be a valuable use of time currently. I also recall hearing whispers that we might move off Heroku in the near-future, which makes me hesitant to spend any more time documenting it than I already have.

## Link to Issue
Links to, but keeps open, #4800.

### CODE REVIEW

**Review by CowMuon — APPROVED** [review]
Fine with this going in as is, OTN would like to see (1) demo environment referenced in context of frick/frack/qa env's so everyone understands what each environment is for. 

(2) Do we have explicit instructions on our traffic control system (updating topic in #eng-infra)? 

(3) As a nice to have, alongside/supplementing how-we-use herroku with what's wrong with heroku for us, documenting the specific limitations & pain points that make us wish to get off PAAS onto IAAS.

**Review by mw2000 — APPROVED** [review]

**Review by Rotorsoft — APPROVED** [review]
LGTM - with ongoing refinements

File: knowledge_base/Testing-Environments.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Testing Environments
 
-Currently we have 2 open testing environments in Heroku for testing any third party integrations requiring a direct connection, called "Frick" and "Frack" and located at https//commonwealth-frick.herokuapp.com and https//commonwealth-frack.herokuapp.com. Frick has access to chain events, which Frack does not. Frack has mainly been used for testing CDN caching (which can't be tested locally) and any sus migrations (that need to be tested without merging into master.)
+Currently we have 2 open testing environments in Heroku for testing any third party integrations requiring a direct connection, called "Frick" and "Frack" and located at `https://commonwealth-frick.herokuapp.com` and `https://commonwealth-frack.herokuapp.com`. Frick has access to chain events, which Frack does not. Frack has mainly been used for testing CDN caching (which can't be tested locally) and any sus migrations (that need to be tested without merging into master.)
```
Comment (timolegros) [inline knowledge_base/Testing-Environments.md:3]: Now that CE 2.0 has been deployed _all_ environments have access to chain-events (Frack included).

Comment (timolegros) [inline knowledge_base/Testing-Environments.md:3]: Now that CE 2.0 has been deployed all environments have access to chain-events (Frack included).

File: knowledge_base/Heroku.md [hunk 2]
```diff
@@ -1,6 +1,66 @@
...
+
+### Common-app
+
+The `common-app` comprises several sub-apps or servers:
+
+1. `commonwealth-beta`: used in the engineering QA process.
+2. `commonwealth-demo`: used by the Design and Product Teams to validate engineer work and provide feedback before merging deploying it to the livesite.
+3. `commonwealth-experimental`: used to host Common 2.0 development.
+4. `commonwealth-frick`: a full Common instance used for testing bots, chain events, and connected services.
+5. `commonwealth-frack`: a standalone raw Common instance that doesn't interact with other services; it is used for testing pure UI.
```
Comment (timolegros) [inline knowledge_base/Heroku.md:37]: Frack can be used for general API or migration testing as well --- main distinction from Frick is that it simply doesn't have connected services.

File: knowledge_base/Testing-Environments.md [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 
-We also have a QA environment which is automatically updated whenever feature branches are merged to master, which is located at <https://commonwealth-beta.herokuapp.com>. However, this QA environment does not currently have access to the chain events service (and also as noted, doesn't have a way of testing migrations without merging them to master, as it is not running on a QA branch.)
+We also have a QA environment which is automatically updated whenever feature branches are merged to master, which is located at `https://commonwealth-beta.herokuapp.com.` However, this QA environment does not currently have access to the chain events service (and also as noted, doesn't have a way of testing migrations without merging them to master, as it is not running on a QA branch.)
```
Comment (timolegros) [inline knowledge_base/Testing-Environments.md]: Same with chain-events here.

File: knowledge_base/Heroku.md [hunk 4]
```diff
@@ -1,6 +1,66 @@
...
+
+The common-bot app several sub-apps or servers:
+
+1. `discobot-listener`: Handles Commonwealth <> Discord integration. For full documentation, see [Discobot.md](./Discobot.md).
+2. `discobot-listener-staging`: A testing server for Discobot.
+3. `farcaster-bot-experimental`: An in-progress Common 2.0 project.
+
+## Procfiles
+
+Heroku procfiles are used to specify the processes (workers, dynos, resources) executed by an app on startup. Processes of the "release" type are run during the [release phase](#release-phases). See [dedicated Heroku documentation](https://devcenter.heroku.com/articles/procfile) for a more detailed guide.
```
Comment (timolegros) [inline knowledge_base/Heroku.md:53]: May be worth mentioning here that we have a `multi-procfile` setup (which also means we use a multi-procfile buildpack): https://github.com/heroku/heroku-buildpack-multi-procfile

File: knowledge_base/Heroku.md [hunk 5]
```diff
@@ -1,6 +1,66 @@
...
+3. `commonwealth-frack-eu`
+4. `cosmos-devnet`
+5. `cosmos-devnet-beta`
+6. `cw-discourse-import`
+7. `evmos-devnet`
+8. `snapshot-listener`
+
+### Common-app
+
+The `common-app` comprises several sub-apps or servers:
```
Comment (timolegros) [inline knowledge_base/Heroku.md:31]: The proper terminology here is that technically `common-app` is a pipeline and then all the apps within it are different stages of the pipeline.

**Review by timolegros — CHANGES_REQUESTED** [review]

**Comment by jnaviask** [comment]
cc @timolegros to re-review, then we can get these changes in

**Review by timolegros — APPROVED** [review]

**Pull request merged**